functors_task_1/student/include/impl.h

```diff
@@ -1,13 +1,27 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <vector>
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    private:
+    int index=1;
+
+    public:
+    BeerBrand operator()()
+    {
+        if(index == (int)BeerBrand::Max)
+        {
+            index = 1;
+        }
+
+        return BeerBrand++;
```

Comment:
Does it compile? You incrementing the enum type name

Reply:
forgot to rebuild, working on it